corehq/apps/data_dictionary/views.py

```diff
@@ -143,6 +147,117 @@ def data_dictionary_json(request, domain, case_type_name=None):
     })
 
 
+@login_and_domain_required
+@requires_privilege_with_fallback(privileges.DATA_DICTIONARY)
+def data_dictionary_json_v2(request, domain, case_type_name=None):
+    fhir_resource_type_name_by_case_type = {}
+    fhir_resource_prop_by_case_prop = {}
+    if toggles.FHIR_INTEGRATION.enabled(domain):
+        fhir_resource_type_name_by_case_type, fhir_resource_prop_by_case_prop = load_fhir_resource_mappings(domain)
+
+    case_type_app_module_count = get_case_type_app_module_count(domain)
+    used_props_by_case_type = get_used_props_by_case_type(domain)
+    geo_case_prop = get_geo_case_property(domain)
+
+    def _get_case_data(case_type):
+        module_count = case_type_app_module_count.get(case_type.name, 0)
+        used_props = used_props_by_case_type.get(case_type.name, [])
+        return {
+            "name": case_type.name,
+            "fhir_resource_type": fhir_resource_type_name_by_case_type.get(case_type),
+            "is_deprecated": case_type.is_deprecated,
+            "module_count": module_count,
+            "is_safe_to_delete": len(used_props) == 0,
+            "properties_count": case_type.properties_count,
+        }
+
+    if case_type_name:
```

Comment:
Nit: This function is quite long. Could we move the code related to fetching data for a single case type into its own separate function? This could even be split up into two separate views.

Reply:
I was in favour of this (moving into a different view) however went against it seeing some amount of common code. I was keen to see what others say during review. 
I am going to split them.

Reply:
+1 for splitting function

Reply:
Yeah, I would also advocate splitting. I would make a few changes:
1. Split `load_fhir_resource_mappings()` into two functions: One for case types and one for case properties.
2. Pull `_get_case_data()` out. Move into it some of the code that would become duplicated. Maybe something like:
   ```
    def get_case_type_data(domain, case_type):
        case_type_app_module_count = get_case_type_app_module_count(domain)
        used_props_by_case_type = get_used_props_by_case_type(domain)
        module_count = case_type_app_module_count.get(case_type.name, 0)
        used_props = used_props_by_case_type.get(case_type.name, [])
        data = {
            "name": case_type.name,
            "is_deprecated": case_type.is_deprecated,
            "module_count": module_count,
            "is_safe_to_delete": len(used_props) == 0,
            "properties_count": case_type.properties_count,
        }
        if toggles.FHIR_INTEGRATION.enabled(domain):
            resource_type_name_by_case_type = get_fhir_resource_type_map(domain)
            data["fhir_resource_type"] = resource_type_name_by_case_type.get(case_type)
        return data
   ```
3. Split the view at the `else:` into two views: One if the case type is given, and one to list the case types.

Reply:
Addressed in 384263b3d074eaa824895851e1f7bdcb4e6b0e94